Source/pfile.cpp

```diff
@@ -225,6 +226,8 @@ bool pfile_ui_set_hero_infos(bool (*ui_add_hero_info)(_uiheroinfo *))
 
 				UnPackPlayer(&pkplr, 0, false);
 
+				pfile_SFileCloseArchive(archive);
+				archive = NULL;
 				LoadHeroItems(&plr[0]);
 				RemoveEmptyInventory(0);
 				CalcPlrInv(0, false);

```

Comment:
Why is there another call `pfile_SFileCloseArchive` on line 238?
If it is needed, it should be removed from there and moved to a more appropriate place

Reply:
If unpacking was a success or not.

Reply:
Because LoadHeroItems opens the archive again and failed earier because it required a closed archive to open it 🤪

Reply:
Instead of calling `close` unconditionally, it should only be called if `open` was called

Reply:
I think i will save that for a later refactor. To much global stuff going on for me to feel safe about shifting stuff around for a hotfix.